views.py

```diff
@@ -68,22 +69,26 @@ def download(request, path, rest_call=False, use_async=True, *args, **kwargs):
 
     resource_cls = check_resource_type(res.resource_type)
 
+    if federated_path:
+        res_root = os.path.join(federated_path, res_id)
+    else:
+        res_root = res_id
+
+    bag_modified = "false"
+    metadata_dirty = "false"
+    if istorage.exists(res_root):
+        bag_modified = istorage.getAVU(res_root, 'bag_modified')
+        metadata_dirty = istorage.getAVU(res_root, 'metadata_dirty')
+
```

Comment:
```bag_modified``` is reset to true whenever there is any metadata change, or there is any file change, like adding or deleting files for a resource, to indicate the bag needs to be regenerated when being downloaded so that all changes to the resource can be picked up in the downloaded bag. In your example, after making changes to metadata, the whole bag will be brought up to date when the science metadata xml file is downloaded by checking this flag and regenerate the bag, then after this download, this flag can be reset to false so that the file will NOT be regenerated next time. As long as you set this flag back to False whenever the bag is brought up to date, regardless whether it is reset when bag is downloaded, or when science metadata xml or resource map xml is downloaded via REST API, or in any other scenarios, as long as the flag is reset properly, the bag will not be regenerated when it does not need to, so I believe this single flag is sufficient and there is no need to introduce another flag.